packages/jsii-pacmak/lib/targets/go/types/go-type.ts

```diff
@@ -189,7 +189,12 @@ export class GoProperty implements GoTypeMember {
         this.getter
       }()${` ${this.returnType}`}`,
     );
-    code.line(`return ${instanceArg}.${this.name}`);
+
+    if (this.parent.name === this.returnType) {
+      code.line(`return *${instanceArg}.${this.name}`);
+    } else {
+      code.line(`return ${instanceArg}.${this.name}`);
```

Comment:
Wait, I'm confused by this -- why is this necessary on emitting the Getter implementation?

Reply:
See https://github.com/aws/jsii/pull/2035/files#r492994719